microsoft/testsuites/vm_extensions/cvt.py

```diff
@@ -0,0 +1,274 @@
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+    def run_script(
+        self, node: Node, log: Logger, test_dir: str
+    ) -> ExecutableResult:
+        timer = create_timer()
+        script: CustomScript = node.tools[self._cvt_script]
+        result = script.run(parameters=test_dir, timeout=19800, sudo=True)
+        log.info(f"Script run with param {test_dir} finished within {timer}")
+        return result
+
+    def copy_cvt_logs(
+        self, log: Logger, node: Node, test_dir: Path, log_path: Path
+    ) -> None:
+        find_tool = node.tools[Find]
+        file_list = find_tool.find_files(
+            test_dir,
+            name_pattern="*.log",
+            file_type="f",
+            sudo=True,
+            ignore_not_exist=True,
+        )
+
+        file_list += find_tool.find_files(
+            test_dir,
+            name_pattern="*.txt",
+            file_type="f",
+            sudo=True,
+            ignore_not_exist=True,
+        )
+
+        for file in file_list:
+            log.info(f"Copying file {file} to {log_path}")
+            try:
+                file_name = file.split("/")[-1]
+                node.shell.copy_back(
+                    node.get_pure_path(file),
+                    log_path / f"{file_name}",
+                )
+            except FileNotFoundError:
+                log.error(f"File {file} doesn't exist.")
+
+    def run_cvt_tests(
+        self,
+        log: Logger,
+        node: Node,
+        log_path: Path,
+        variables: Dict[str, Any],
+    ) -> Optional[int]:
+        cvt_bin = "indskflt_ct"
+        sas_uri = variables.get("cvt_binary_sas_uri", "")
+        if not sas_uri:
+            log.error("sas uri for cvt binary is empty.")
+            return 1
+        cvt_root_dir = str(node.working_path) + "/LisaTest/"
+        cvt_download_dir = cvt_root_dir + "cvt_files/"
+
+        mkdir = node.tools[Mkdir]
+        wget = node.tools[Wget]
+
+        mkdir.create_directory(cvt_download_dir, sudo=True)
+        download_path = wget.get(
+            url=f"{sas_uri}",
+            filename=cvt_bin,
+            file_path=cvt_download_dir,
+            sudo=True,
+        )
+
+        cvt_md5sum = node.execute(
+            f"md5sum {download_path}", shell=True, sudo=True
+        )
+        log.info(f"md5sum '{download_path}' : '{cvt_md5sum}'")
+
+        result = self.run_script(node=node, log=log, test_dir=cvt_download_dir)
+        self.copy_cvt_logs(
+            node=node,
+            log=log,
+            test_dir=Path(node.working_path.parent.parent),
+            log_path=log_path,
+        )
+        return result.exit_code
+
+    @TestCaseMetadata(
+        description="""
+        this test validate the functionality of ASR driver by verifying
+        integrity of a source disk with respect to a target disk
+        """,
+        priority=1,
+        use_new_environment=True,
+        timeout=TIMEOUT,
+    )
+    def run_cvt(
```

Comment:
1. Rename it like `verify_asr_by_cvt`.
2. Move it to top of the test suite, so it's easy to see test cases in a test suite. For other methods, only `before_case` should be in the test suite. Other methods should be private and as function in this file, not method in the test suite class.

Reply:
1. Renamed the method
2. Moved the method to functions in file